src/components/product/Products.js

```diff
@@ -19,7 +22,7 @@ const Products = () => {
 
   const getProductUpdate = (id) => {
     const product = state.product.filter((prod) => prod.id === id);
-    setProduct({ ...product });
+    setProduct(...product);
   };
```

Comment:
the state already has a name called product, here use one more time create const product...., it would be cause concern here, change the name to something like: `prdFiltered`